src/main/java/org/example/services/JobRoleService.java

```diff
@@ -14,18 +14,35 @@
 public class JobRoleService {
     private final JobRoleDao jobRoleDao;
     private final JobRoleMapper jobRoleMapper;
+    private final int maxPageSize = 10;
 
     public JobRoleService(final JobRoleDao jobRoleDao,
                           final JobRoleMapper jobRoleMapper) {
         this.jobRoleDao = jobRoleDao;
         this.jobRoleMapper = jobRoleMapper;
     }
 
-    public List<JobRoleResponse> getAllJobRoles() throws SQLException {
-
-        return jobRoleMapper.mapJobRoleListToJobRoleResponseList(
-                jobRoleDao.getJobRoles());
+    public List<JobRoleResponse> getAllJobRoles(
+            final int page, final int pageSize) throws SQLException {
+        if (page <= 0 || pageSize <= 0) {
```

Comment:
validate each of the fields separately...

SUGGESTION
if (page <=0) { page = 1; }
if (page > (total recs / pageSize) { page = 'last page' }



if (pageSize < MIN_PAGE_SIZE) { pageSize  = MIN_PAGE_SIZE; } 
if (pageSize > MAX_PAGE_SIZE) { pageSize  = MAX_PAGE_SIZE; }